test/mako/alerter/slack/message.go

```diff
@@ -85,12 +86,18 @@ func (smh *MessageHandler) SendAlert(testName, summary string) error {
 			); err != nil {
 				errCh <- fmt.Errorf("failed to retrieve message history in channel %q", channel.Name)
 			}
-			// do not send message again if messages were sent on the same channel a short while ago
-			if len(messageHistory) != 0 {
-				return
+			// decorate the test name for more accurate match
+			decoratedTestName := decoratedName(testName)
+			// do not send message again if alert for this test has been sent to
+			// the channel a short while ago
+			for _, message := range messageHistory {
+				if strings.Contains(message, decoratedTestName) {
```

Comment:
do you need to worry about backward compatible?

Reply:
The last alert was sent **two weeks** ago, and the reason we check the message history is we don't want to sent to many alerts within a short time, which is set to **24 hours** here. So  it won't be impacted by the previously alert messages.